src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -590,3 +662,51 @@ sol_http_server_set_last_modified(struct sol_http_server *server, const char *pa
 
     return -ENODATA;
 }
+
+SOL_API int
+sol_http_server_add_dir(struct sol_http_server *server, const char *rootdir)
+{
+    int r;
+    uint16_t i;
+    char *dir;
+
+    SOL_NULL_CHECK(server, -EINVAL);
+    SOL_NULL_CHECK(rootdir, -EINVAL);
+
+    SOL_PTR_VECTOR_FOREACH_IDX (&server->dirs, dir, i) {
+        if (streq(dir, rootdir))
+            return -EINVAL;
+    }
+
+    dir = strdup(rootdir);
```

Comment:
Instead of `strdup()`, `realpath()`, as that's what's being compared in the function that opens the file.